app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,17 @@
+class RecipesController < ApplicationController
+
+  def search; end
+
+  def index
+    @query = params[:query]
+    search = EdamamApiWrapper.search(@query)
+    if search
+      @recipes = search.recipes.paginate(page: params[:page], per_page: 12)
+    end
+  end
+
+  def show
+    @recipe = EdamamApiWrapper.single_search(params[:id])
+  end
```

Comment:
What if the search fails? Your page should probably report the failure to the user.